_setup_support.py

```diff
@@ -152,6 +154,12 @@ def summarize(self) -> None:
         kind = "NEWLY BUILT" if self.action == "build" else "PREVIOUSLY BUILT"
         print(SUMMARY_MSG.format(kind=kind))
 
+class DevelopWithJs(develop):
+    def run(self):
+        super().run()
+        if not self.uninstall:
+            self.run_command("build_js")
```

Comment:
The downside of the approach adding a new subcommand instead of replacing `build_ext` is that `pip install -e .` will not run this subcommand by default.

Overwriting the `develop` command to make it run `build_js` should handle this scenario.

It is important however to notice that this solution is valid while PEP 660 is not implemented in setuptools. In the [near future](https://github.com/pypa/setuptools/pull/3265), `setuptools` might add an API so `build` subcommands can tell which files to install...

It is very likely that this new API will require subcommand authors to implement methods like:
```python
def get_outputs(self) -> List[str]:
    """Return the list of files created by the build command"""

# and/or
def get_editable_mapping(self) -> Dict[str, str]:
    """Return a directory mapping files existing on the project tree/sdist to files as they would be present in the wheel"""
```
but I am not sure yet...

Reply:
What is the downside to replacing `build_ext`? Unless I am mistaken it happens for for `install .` and `install -e .` (which is what we need) and it seems s simpler way to get to that.